### PR TITLE
Bluetooth: host: don't timeout when reconfiguring an l2cap channel

### PR DESCRIPTION
**Pull request by jori-nordic**
The request timer started on the first l2cap channel when doing an MTU
reconfiguration has to be stopped when the response is received. Else the
channel will get into a bad state.

Signed-off-by: Jonathan Rico <jonathan.rico@nordicsemi.no>

### CODE REVIEW

**Review by alwa-nordic — CHANGES_REQUESTED** [review]
As discussed f2f, perform the rtx cancel on all the ch matching ident, and add the discussed comments in the _req function and here.

File: subsys/bluetooth/host/l2cap.c [hunk 1]
```diff
@@ -1419,6 +1424,8 @@ static void le_ecred_reconf_rsp(struct bt_l2cap *l2cap, uint8_t ident,
 			ch->chan.ops->reconfigured(&ch->chan);
 		}
 	}
+
+	BT_DBG("");
```
Comment (Thalley) [inline subsys/bluetooth/host/l2cap.c:1428]: Remove?

**Review by alwa-nordic — DISMISSED** [review]

File: subsys/bluetooth/host/l2cap.c (hunk 1 above)
Comment (jori-nordic) [reply subsys/bluetooth/host/l2cap.c:1428]: I added this to get the function name in the log, making sure we process the response. I can remove it if you really want.

**Review by alwa-nordic — APPROVED** [review]

**Comment by jori-nordic** [comment]
@Thalley can you approve if you don't have any more comments ?

**Review by Thalley — APPROVED** [review]

**Pull request merged**